src/image/descriptor.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+make_pub!(
+    #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+    #[cfg_attr(
+        feature = "builder",
+        derive(derive_builder::Builder, getset::Getters),
+        builder(pattern = "owned", setter(into, strip_option)),
+        getset(get = "pub")
+    )]
+    /// A Content Descriptor (or simply Descriptor) describes the disposition of the targeted content.
+    /// It includes the type of the content, a content identifier (digest), and the byte-size of the raw content.
+    /// Descriptors SHOULD be embedded in other formats to securely reference external content.
+    struct Descriptor {
+        /// This REQUIRED property contains the media type of the referenced
+        /// content. Values MUST comply with RFC 6838, including the naming
+        /// requirements in its section 4.2.
+        #[serde(rename = "mediaType")]
+        media_type: String,
```

Comment:
```suggestion
        #[cfg_attr(feature = "builder", getset(get = "pub"))]
        media_type: String,
```
as well as for the others below (but not for `size: i64`)